docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,95 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices OSDs so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    Kind: StorageAutoScaling
+    Metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    Spec:
+        storageClusterName: <storageclusterName>
+        deviceClassName: <deviceClassName>
+        capacityLimit: <capacity>
+        maxOSDSize: 8Ti
```

Comment:
The latest best-practice I'm aware of is to use CamelCase even for acronyms like OSD.

```suggestion
        maxOsdSize: 8Ti
```